src/main/java/io/kestra/plugin/langchain/OpenAIChatMemory.java

```diff
@@ -0,0 +1,159 @@
+package io.kestra.plugin.langchain;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.UserMessage;
+import dev.langchain4j.memory.ChatMemory;
+import dev.langchain4j.memory.chat.TokenWindowChatMemory;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.openai.OpenAiChatModel;
+import dev.langchain4j.model.openai.OpenAiChatModelName;
+import dev.langchain4j.model.openai.OpenAiTokenizer;
+import dev.langchain4j.store.memory.chat.InMemoryChatMemoryStore;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.property.Property;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.langchain.exceptions.ApiKeyNotFoundException;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import org.slf4j.Logger;
+import jakarta.validation.constraints.NotNull;
+
+import java.util.List;
+import java.util.UUID;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "LangChain4j Chat Memory Task",
+    description = "Handles chat interactions with memory using LangChain4j"
+)
+@Plugin(
+    examples = {
+        @io.kestra.core.models.annotations.Example(
+            title = "Chat Memory Example",
+            code = {
+                "prompt: \"Hello, my name is John\"",
+                "model: \"gpt-4\"",
+                "maxTokens: 300"
+            }
+        )
+    }
+)
+public class OpenAIChatMemory extends Task implements RunnableTask<OpenAIChatMemory.Output> {
+
+    @Schema(
+        title = "User message",
+        description = "The input message from the user"
+    )
+    @NotNull
+    private Property<String> userMessage;
+
+    @Schema(
+        title = "API Key",
+        description = "OpenAI API key"
+    )
+    @NotNull
+    private Property<String> apiKey;
+
+    @Schema(
+        title = "OpenAI Model",
+        description = "OpenAI model name"
+    )
+    private Property<OpenAiChatModelName> modelName;
+
+    @Schema(
+        title = "Max Tokens",
+        description = "Maximum tokens for chat memory"
+    )
+    private Property<Integer> maxTokens;
+
+    @Schema(
+        title = "Chat Memory ID",
+        description = "The unique ID for the chat memory"
+    )
+    private Property<UUID> chatMemoryId;
+
+    private ChatMemory chatMemory;
+
+    @Override
+    public OpenAIChatMemory.Output run(RunContext runContext) throws Exception {
+        Logger logger = runContext.logger();
+
+        // Render input properties
+        String renderedUserMessage = runContext.render(userMessage).as(String.class)
+            .orElseThrow(() -> new IllegalArgumentException("User message is required"));
```

Comment:
```suggestion
            .orElseThrow();
```
Not need to add a meaningfull error message as the property is mandatory (annotated with `@NotNull`) it will always be set as you cannot save a flow with validation error.

Please do the same for in all other places.